src/features.c

```diff
@@ -0,0 +1,344 @@
+}
+
+void
+crun_features_add_cgroup_info (yajl_gen json_gen, const struct linux_info_s *linux)
+{
+  yajl_gen_string (json_gen, (const unsigned char *) "cgroup", strlen ("cgroup"));
+  yajl_gen_map_open (json_gen);
+
+  add_bool_to_json (json_gen, "v1", linux->cgroup.v1);
+  add_bool_to_json (json_gen, "v2", linux->cgroup.v2);
+  add_bool_to_json (json_gen, "systemd", linux->cgroup.systemd);
+  add_bool_to_json (json_gen, "systemdUser", linux->cgroup.systemdUser);
+
+  yajl_gen_map_close (json_gen);
+}
+
+void
+crun_features_add_seccomp_info (yajl_gen json_gen, const struct linux_info_s *linux)
+{
+  yajl_gen_string (json_gen, (const unsigned char *) "seccomp", strlen ("seccomp"));
+  yajl_gen_map_open (json_gen);
+
+  add_bool_to_json (json_gen, "enabled", linux->seccomp.enabled);
+  add_array_to_json (json_gen, "actions", linux->seccomp.actions);
+  add_array_to_json (json_gen, "operators", linux->seccomp.operators);
+  add_array_to_json (json_gen, "archs", linux->seccomp.archs);
+
+  yajl_gen_map_close (json_gen);
+}
+
+void
+crun_features_add_apparmor_info (yajl_gen json_gen, const struct linux_info_s *linux)
+{
+  yajl_gen_string (json_gen, (const unsigned char *) "apparmor", strlen ("apparmor"));
+  yajl_gen_map_open (json_gen);
+
+  add_bool_to_json (json_gen, "enabled", linux->apparmor.enabled);
+
+  // Close the map for apparmor
+  yajl_gen_map_close (json_gen);
+}
+
+void
+crun_features_add_selinux_info (yajl_gen json_gen, const struct linux_info_s *linux)
+{
+  yajl_gen_string (json_gen, (const unsigned char *) "selinux", strlen ("selinux"));
+  yajl_gen_map_open (json_gen);
+
+  add_bool_to_json (json_gen, "enabled", linux->selinux.enabled);
+
+  // Close the map for selinux
+  yajl_gen_map_close (json_gen);
+}
+
+void
+crun_features_add_linux_info (yajl_gen json_gen, const struct linux_info_s *linux)
+{
+  yajl_gen_string (json_gen, (const unsigned char *) "linux", strlen ("linux"));
+  yajl_gen_map_open (json_gen);
+
+  crun_features_add_namespaces (json_gen, linux);
+  crun_features_add_capabilities (json_gen, linux);
+  crun_features_add_cgroup_info (json_gen, linux);
+  crun_features_add_seccomp_info (json_gen, linux);
+  crun_features_add_apparmor_info (json_gen, linux);
+  crun_features_add_selinux_info (json_gen, linux);
+
+  yajl_gen_map_close (json_gen);
+}
+
+void
+crun_features_add_annotations_info (yajl_gen json_gen, const struct annotations_info_s *annotation)
+{
+  yajl_gen_string (json_gen, (const unsigned char *) "annotations", strlen ("annotations"));
+  yajl_gen_map_open (json_gen);
+
+  add_string_to_json (json_gen, "io.github.seccomp.libseccomp.version", annotation->io_github_seccomp_libseccomp_version);
+  add_bool_to_json (json_gen, "org.opencontainers.runc.checkpoint.enabled", annotation->org_opencontainers_runc_checkpoint_enabled);
+  add_string_to_json (json_gen, "io.github.containers.crun.commit", annotation->io_github_containers_crun_commit);
```

Comment:
let's use the prefix `run.oci` instead of `io.github.containers.crun`.

If using something more generic than `org.opencontainers.runc` is not possible, then add another variable with the `run.oci` prefix too and store there the same value

Reply:
so I was planning to use https://github.com/containers/crun/pull/1237#discussion_r1239922283, however, per [this](https://github.com/containers/crun/pull/1237#discussion_r1239936681) suggestion, it seems reasonable to go with the runc annotation. I can add another field with the `run.oci` if that works for you.